### DISCUSSION TITLE
`Arr::unwrap` and `single` methods

### DISCUSSION THREAD

**Discussion by michael-rubel** [laravel/framework#42205] [Ideas]
Hi, Laravel Folks!

I always miss these features when building packages around Laravel:
- Ability to identify if an array/countable has only one element, al'a `count === 1` but with some beautiful syntax method, for example, `single`, or `one`.
- Kind of `Arr::unwrap` method, to return value from an array if it contains only one key inside.

I thought about PRs, but still not sure it is worth an effort and wouldn't be rejected. What do you think?

**Comment by sitroz-oz** [comment]
I need this function too. I think it migth be more elegant and helpful.

Function code:
```php
    public static function unwrap($value)
    {
        if (is_countable($value) && count($value) === 1) {
            return self::first($value);
        }

        return $value;
    }
```

Usage:
```php
    /**
     * @param string|string[]|Collection $input single or many 
     * @return string|string[] or smth else
     */
    public function foo(string|array|Collection $input): string|array
    {
        $output = [];
        
        foreach (Arr::wrap($input) as $item) {
            
            // Do smth
            $output[$item] = 'bar';
        }
        
        // if (count($output) === 1) {
        //     return Arr::first($output);
        // }
        //
        // return $output;
        
        return Arr::unwrap($output); 
    }
```
